lib/navigation/app_controller.dart

```diff
@@ -41,5 +57,14 @@ class _MyAppState extends State<AppController> {
         accentColor: Color(0xFF45cab9),
       )
     );
+    
+    signOut() async {
+    try {
+      await widget.auth.signOut();
+      widget.logoutCallback();
+    } catch (e) {
+      print(e);
+    }
+  }
```

Comment:
This whole `signOut()` function is dead code and is not being reached. To fix it, you have to move it outside of the `Widget build(BuildContext context)` function. Also make sure to add an `on` clause before `catch`.
```
signOut() async {
    try {
      await widget.auth.signOut();
      widget.logoutCallback();
    } on Exception catch (e) {
      print(e);
    }
  }
```

Reply:
I see.